File: 18/main.py
def fourSum(nums, target):
  nums = sorted(nums)
  result = []
  memory = []
  for i in range(0, len(nums) - 3):
    for j in range(i + 1, len(nums) - 2):
      for k in range(j + 1, len(nums) - 1):
        for l in range(k + 1, len(nums)):
          if (nums[i] + nums[j] + nums[k] + nums[l] == target and str([nums[i], nums[j], nums[k], nums[l]]) not in memory):
            memory.append(str([nums[i], nums[j], nums[k], nums[l]]))
            result.append([nums[i], nums[j], nums[k], nums[l]])
  return result


# TEST SUITE

# TEST 1

print(fourSum([1, 0, -1, 0, -2, 2], 0))

# TEST 2

print(fourSum([-3,-2,-1,0,0,1,2,3], 0))

File: 18/main.js
var fourSum = function(nums, target) {
  nums = nums.sort((a,b) => a - b);
  let result = new Set();
  for (let i = 0; i < nums.length - 3; i++) {
    for (let j = i + 1; j < nums.length - 2; j++) {
      for (let k = j + 1; k < nums.length - 1; k++) {
        for (let l = k + 1; l < nums.length; l++) {
          if (nums[i] + nums[j] + nums[k] + nums[l] === target) {
            result.add(JSON.stringify([nums[i], nums[j], nums[k], nums[l]]))
          }
        }
      }
    }
  }
  result = Array.from(result);
  result = result.map(string => JSON.parse(string));
  return result;
};


// TEST SUITE

// TEST 1

console.log(fourSum([1, 0, -1, 0, -2, 2], 0))

// TEST 2

console.log(fourSum([-3,-2,-1,0,0,1,2,3], 0))